src/main/java/com/example/crm_gym/controllers/TraineeController.java

```diff
@@ -0,0 +1,365 @@
+    @ApiOperation(value = "Get a trainee's trainings", response = Trainee.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved the trainee's trainings."),
+            @ApiResponse(code = 400, message = "Invalid input data."),
+            @ApiResponse(code = 404, message = "No trainings found for the given criteria."),
+            @ApiResponse(code = 500, message = "Application failed to process the request.")
+    })
+    public ResponseEntity<?> getTraineeTrainings(
+            @Valid @RequestParam("username") String username,
+            @RequestParam(name = "fromDate", required = false) @DateTimeFormat(pattern = "yyyy-MM-dd") Date fromDate,
+            @RequestParam(name = "toDate", required = false) @DateTimeFormat(pattern = "yyyy-MM-dd") Date toDate,
+            @Valid @RequestParam(name = "trainerName", required = false) String trainerName,
+            @Valid @RequestParam(name = "trainingType", required = false) String trainingTypeName,
+            HttpServletRequest request) {
+        String transactionId = TransactionLogger.generateTransactionId();
+        TransactionLogger.logTransactionStart(transactionId, "GetTrainee's Trainings");
+        TransactionLogger.logRequestDetails(transactionId, request.getMethod(), request.getRequestURI(), request.getParameterMap());
+
+        Optional<List<Training>> optionalTrainings = traineeService.getTrainingsByTraineeUsernameAndCriteria(username, fromDate, toDate, trainerName, trainingTypeName, transactionId);
+
+        if (!optionalTrainings.isPresent() || optionalTrainings.get().isEmpty()) {
+            TransactionLogger.logResponseDetails(transactionId, HttpStatus.NOT_FOUND.value(), "Get Trainee Trainings Failed");
+            TransactionLogger.logTransactionEnd(transactionId, "Get Trainee Trainings");
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body(Collections.singletonMap("error", "No trainings found for the given criteria"));
+        }
+
+        List<TrainingDTO> trainingDTOList = optionalTrainings.get().stream().map(training -> {
+
+            TrainerDTO trainerDTO = new TrainerDTO(
+                    new UserDTO(
+                            training.getTrainer().getUser().getUsername()
+                    )
+            );
+
+            return new TrainingDTO(
+                    training.getTrainingName(),
+                    training.getTrainingDate(),
+                    new TrainingTypeDTO(training.getTrainingType().getName()),
+                    training.getTrainingDuration(),
+                    trainerDTO
+            );
+        }).collect(Collectors.toList());
+
+        TransactionLogger.logResponseDetails(transactionId, HttpStatus.OK.value(), "Get Trainee Trainings");
+        TransactionLogger.logTransactionEnd(transactionId, "Get Trainee Trainings");
+        return ResponseEntity.ok(trainingDTOList);
+    }
+
+    @GetMapping("/active-trainers")
```

Comment:
I would recommend not to create separate path for active-trainers.
GET /trainers and RequestParam=isActive and pass here true or false